### PR TITLE
wagmi hook generator with examples

### PR DESCRIPTION
**Pull request by Amirjab21**

### CODE REVIEW

File: packages/greenfield-app/pages/_app.tsx [hunk 1]
```diff
@@ -20,22 +21,18 @@ import "../styles/globals.css";
...
 
 const bnb: Chain = {
-  id: 56,
-  name: "BNB Chain",
-  network: "bnb",
-  iconUrl: "https://assets.coingecko.com/coins/images/825/large/binance-coin-logo.png?1547034615",
-  rpcUrls: { default: "https://bsc-dataseed1.binance.org" },
-  blockExplorers: { default: { name: "BSCScan", url: "https://bscscan.com" } },
+  ...bsc,
+  rpcUrls: { default: { http: ["https://bsc-dataseed1.binance.org"] }, public: { http: ["https://bsc-dataseed1.binance.org"] } },
```
Comment (RedVeil) [inline packages/greenfield-app/pages/_app.tsx:25]: if bsc is now supported we can get rid of the whole config here

File: packages/greenfield-app/pages/_app.tsx [hunk 2]
```diff
@@ -3,15 +3,16 @@ import { FeatureTogglePanel } from "@popcorn/components/components/FeatureToggle
 import { DualActionModalContainer } from "@popcorn/components/components/Modal/DualActionModalContainer";
 import { MultiChoiceActionModalContainer } from "@popcorn/components/components/Modal/MultiChoiceActionModalContainer";
 import { SingleActionModalContainer } from "@popcorn/components/components/Modal/SingleActionModalContainer";
-import OfacCheck from "@popcorn/app/components/OfacCheck";
+import OfacCheck from "../../app/components/OfacCheck";
```
Comment (RedVeil) [inline packages/greenfield-app/pages/_app.tsx:6]:
```suggestion
import OfacCheck from "@popcorn/app/components/OfacCheck";
```

File: packages/greenfield-app/package.json [hunk 3]
```diff
@@ -4,10 +4,11 @@
   "private": true,
   "scripts": {
     "dev": "next dev",
-    "build": "next build",
+    "build": "next build && yarn wagmi generate",
```
Comment (RedVeil) [inline packages/greenfield-app/package.json:7]: should we not generate first? in case we use these hooks later for the build?

Comment (RedVeil) [inline packages/greenfield-app/package.json:7]:
```suggestion
    "build": "yarn wagmi generate && next build",
```

File: packages/greenfield-app/pages/_app.tsx (hunk 1 above)
Comment (Amirjab21) [reply packages/greenfield-app/pages/_app.tsx:25]: alright, I was just keeping it in incase we were using a different rpcURL from the default

File: packages/greenfield-app/package.json (hunk 3 above)
Comment (Amirjab21) [reply packages/greenfield-app/package.json:7]: great spot

**Pull request merged**